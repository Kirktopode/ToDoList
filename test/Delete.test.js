import React from 'react'
import { render } from 'react-dom'
import { Provider } from 'react-redux'
import { createStore } from 'redux'
import { addTodo,
         deleteTodo
} from '../src/actions'
import todoApp from '../src/reducers'
import App from '../src/components/App'
import renderer from 'react-test-renderer';

test('When the second item from a list of ToDo items is deleted, all of the items remain except for the one that was deleted.', () =>{

    //Mocking Date for the sake of testing
    global.Date = jest.fn(() => {return {
      toLocaleDateString: () => {
        return "test date"
      },
      toLocaleTimeString: () => {
        return "test time"
      }
    }})
    let store = createStore(todoApp)
    store.dispatch(addTodo("Test0"))
    store.dispatch(addTodo("Test1"))
    store.dispatch(addTodo("Test2"))
    store.dispatch(addTodo("Test3"))
    store.dispatch(addTodo("Test4"))
    store.dispatch(addTodo("Test5"))
    var container = document.createElement('div')
    let component = renderer.create(
      <Provider store={store}>
        <App />
      </Provider>,
      container
    );

    let tree = component.toJSON()
    /*global.document = jest.fn(() => { return {
      getElementById: () => {
        return tree.children[1].children[0].children[3].props
      }
    }})
  */
    expect(tree.children[1].children.length).toBe(6)
    expect(tree).toMatchSnapshot()

    //This line works just fine, but replacing click event with dispatch
    //tree.children[1].children[1].children[0].props.onClick()
    store.dispatch(deleteTodo(1))

    tree = component.toJSON()
    expect(tree).toMatchSnapshot()
    expect(tree.children[1].children[0].children[2].children[0]).toMatch(/^Test0  -/)
    expect(tree.children[1].children[1].children[2].children[0]).toMatch(/^Test2  -/)
    expect(tree.children[1].children[2].children[2].children[0]).toMatch(/^Test3  -/)
  })
