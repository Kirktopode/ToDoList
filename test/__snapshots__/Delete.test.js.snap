// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When the second item from a list of ToDo items is deleted, all of the items remain except for the one that was deleted. 1`] = `
<div>
  <div>
    <form
      onSubmit={[Function]}
    >
      <input />
      <button
        type="submit"
      >
        Add Todo
      </button>
    </form>
  </div>
  <ul>
    <li
      style={
        Object {
          "textDecoration": "none",
        }
      }
    >
      <button
        onClick={[Function]}
      >
        x
      </button>
      <button
        onClick={[Function]}
      >
        Append Time
      </button>
      <span
        onClick={[Function]}
      >
        Test0  - Created test time test date
      </span>
      <input
        id="copy-count-0"
        type="number"
      />
      <button
        onClick={[Function]}
      >
        Copy
      </button>
    </li>
    <li
      style={
        Object {
          "textDecoration": "none",
        }
      }
    >
      <button
        onClick={[Function]}
      >
        x
      </button>
      <button
        onClick={[Function]}
      >
        Append Time
      </button>
      <span
        onClick={[Function]}
      >
        Test1  - Created test time test date
      </span>
      <input
        id="copy-count-1"
        type="number"
      />
      <button
        onClick={[Function]}
      >
        Copy
      </button>
    </li>
    <li
      style={
        Object {
          "textDecoration": "none",
        }
      }
    >
      <button
        onClick={[Function]}
      >
        x
      </button>
      <button
        onClick={[Function]}
      >
        Append Time
      </button>
      <span
        onClick={[Function]}
      >
        Test2  - Created test time test date
      </span>
      <input
        id="copy-count-2"
        type="number"
      />
      <button
        onClick={[Function]}
      >
        Copy
      </button>
    </li>
    <li
      style={
        Object {
          "textDecoration": "none",
        }
      }
    >
      <button
        onClick={[Function]}
      >
        x
      </button>
      <button
        onClick={[Function]}
      >
        Append Time
      </button>
      <span
        onClick={[Function]}
      >
        Test3  - Created test time test date
      </span>
      <input
        id="copy-count-3"
        type="number"
      />
      <button
        onClick={[Function]}
      >
        Copy
      </button>
    </li>
    <li
      style={
        Object {
          "textDecoration": "none",
        }
      }
    >
      <button
        onClick={[Function]}
      >
        x
      </button>
      <button
        onClick={[Function]}
      >
        Append Time
      </button>
      <span
        onClick={[Function]}
      >
        Test4  - Created test time test date
      </span>
      <input
        id="copy-count-4"
        type="number"
      />
      <button
        onClick={[Function]}
      >
        Copy
      </button>
    </li>
    <li
      style={
        Object {
          "textDecoration": "none",
        }
      }
    >
      <button
        onClick={[Function]}
      >
        x
      </button>
      <button
        onClick={[Function]}
      >
        Append Time
      </button>
      <span
        onClick={[Function]}
      >
        Test5  - Created test time test date
      </span>
      <input
        id="copy-count-5"
        type="number"
      />
      <button
        onClick={[Function]}
      >
        Copy
      </button>
    </li>
  </ul>
  <p>
    Show:
     
    <span>
      All
    </span>
    , 
    <a
      href=""
      onClick={[Function]}
    >
      Active
    </a>
    , 
    <a
      href=""
      onClick={[Function]}
    >
      Completed
    </a>
    <input
      checked={true}
      onChange={[Function]}
      type="checkbox"
    />
    Time
  </p>
</div>
`;

exports[`When the second item from a list of ToDo items is deleted, all of the items remain except for the one that was deleted. 2`] = `
<div>
  <div>
    <form
      onSubmit={[Function]}
    >
      <input />
      <button
        type="submit"
      >
        Add Todo
      </button>
    </form>
  </div>
  <ul>
    <li
      style={
        Object {
          "textDecoration": "none",
        }
      }
    >
      <button
        onClick={[Function]}
      >
        x
      </button>
      <button
        onClick={[Function]}
      >
        Append Time
      </button>
      <span
        onClick={[Function]}
      >
        Test0  - Created test time test date
      </span>
      <input
        id="copy-count-0"
        type="number"
      />
      <button
        onClick={[Function]}
      >
        Copy
      </button>
    </li>
    <li
      style={
        Object {
          "textDecoration": "none",
        }
      }
    >
      <button
        onClick={[Function]}
      >
        x
      </button>
      <button
        onClick={[Function]}
      >
        Append Time
      </button>
      <span
        onClick={[Function]}
      >
        Test2  - Created test time test date
      </span>
      <input
        id="copy-count-1"
        type="number"
      />
      <button
        onClick={[Function]}
      >
        Copy
      </button>
    </li>
    <li
      style={
        Object {
          "textDecoration": "none",
        }
      }
    >
      <button
        onClick={[Function]}
      >
        x
      </button>
      <button
        onClick={[Function]}
      >
        Append Time
      </button>
      <span
        onClick={[Function]}
      >
        Test3  - Created test time test date
      </span>
      <input
        id="copy-count-2"
        type="number"
      />
      <button
        onClick={[Function]}
      >
        Copy
      </button>
    </li>
    <li
      style={
        Object {
          "textDecoration": "none",
        }
      }
    >
      <button
        onClick={[Function]}
      >
        x
      </button>
      <button
        onClick={[Function]}
      >
        Append Time
      </button>
      <span
        onClick={[Function]}
      >
        Test4  - Created test time test date
      </span>
      <input
        id="copy-count-3"
        type="number"
      />
      <button
        onClick={[Function]}
      >
        Copy
      </button>
    </li>
    <li
      style={
        Object {
          "textDecoration": "none",
        }
      }
    >
      <button
        onClick={[Function]}
      >
        x
      </button>
      <button
        onClick={[Function]}
      >
        Append Time
      </button>
      <span
        onClick={[Function]}
      >
        Test5  - Created test time test date
      </span>
      <input
        id="copy-count-4"
        type="number"
      />
      <button
        onClick={[Function]}
      >
        Copy
      </button>
    </li>
  </ul>
  <p>
    Show:
     
    <span>
      All
    </span>
    , 
    <a
      href=""
      onClick={[Function]}
    >
      Active
    </a>
    , 
    <a
      href=""
      onClick={[Function]}
    >
      Completed
    </a>
    <input
      checked={true}
      onChange={[Function]}
      type="checkbox"
    />
    Time
  </p>
</div>
`;
